tests/tests-core/src/main/java/com/gentics/mesh/search/ElasticSearchProviderTimeoutTest.java

```diff
@@ -47,18 +59,50 @@ public static void setupTimeoutServer() throws IOException {
 		ElasticSearchOptions options = testContext.getOptions().getSearchOptions();
 		String url = options.getUrl();
 		int port = new URL(url).getPort();
+		realClient = WebClient.create(vertx);
 		// Create a dummy server which just blocks on every in-bound request for 1 second
-		HttpServer server = vertx.createHttpServer(new HttpServerOptions().setPort(port));
+		server = vertx.createHttpServer(new HttpServerOptions().setPort(port));
 		server.requestHandler(rh -> {
-			// Don't block validation requests.
-			if (rh.absoluteURI().indexOf("_template/validation") > 0) {
-				rh.response().end(new JsonObject().encodePrettily());
-			} else {
-				log.info("Waiting for 16 second to answer request: " + rh.absoluteURI());
-				vertx.setTimer(Duration.ofSeconds(3).toMillis(), th -> rh.response().end());
+			try {
+				if (timeout) {
```

Comment:
The request is proxied to the original ES (since it's running anyway), being wrapped into the functionality we need (timeout / custom response).